File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    let losted = lost.sort((a, b) => a-b).filter((x) => !reserve.includes(x));
    let reserved = reserve.sort((a, b) => a-b).filter((x) => !lost.includes(x));
    
    const result = losted.filter((x) => {
        const rent = reserved.find((y) => Math.abs(y-x)===1);
        
        if(!rent) return x;
        reserved = reserved.filter((y) => y!==rent);
    });
    
    return n-result.length;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    reserved = sorted(filter(lambda x: x not in lost, reserve))
    lost = sorted(filter(lambda x: x not in reserve, lost))

    for r in reserved:
        if r - 1 in lost:
            lost.remove(r - 1)
        elif r + 1 in lost:
            lost.remove(r + 1)

    return n - len(lost)